edi/models/edi_backend.py

```diff
@@ -37,3 +47,83 @@ def _get_component(self, safe=False, work_ctx=None, **kw):
+        :param values: edi.exchange.record values
+        :return: edi.exchange.record record
+        """
+        self.ensure_one()
+        _values = self._create_record_prepare_values(type_code, values)
+        return self.env["edi.exchange.record"].create(_values)
+
+    def _create_record_prepare_values(self, type_code, values):
+        res = values.copy()  # do not pollute original dict
+        export_type = self.env["edi.exchange.type"].search(
+            self._get_exchange_type_domain(type_code), limit=1
+        )
+        export_type.ensure_one()
+        res["type_id"] = export_type.id
+        res["backend_id"] = self.id
+        return res
+
+    def _get_exchange_type_domain(self, code):
+        return [
+            ("code", "=", code),
+            "|",
+            ("backend_type_id", "=", self.backend_type_id.id),
+            ("backend_id", "=", self.id),
+        ]
+
+    def generate_output(self, exchange_record, store=True, force=False, **kw):
+        """Generate output content for given exchange record.
+
+        :param exchange_record: edi.exchange.record recordset
+        :param store: store output on the record itself
+        :param force: allow to re-genetate the content
+        :param kw: keyword args to be propagated to output generate handler
+        """
+        self.ensure_one()
+        self._validate_generate_output(exchange_record, force=force)
+        output = self._generate_output(exchange_record, **kw)
+        if output and store:
+            if not isinstance(output, bytes):
+                output = output.encode()
+            exchange_record.update(
+                {
+                    "exchange_file": base64.b64encode(output),
+                    "edi_exchange_state": "output_pending",
+                }
+            )
+        return tools.pycompat.to_text(output)
+
+    def _validate_generate_output(self, exchange_record, force=False):
+        exchange_record.ensure_one()
+        if (
+            exchange_record.edi_exchange_state != "new"
+            and exchange_record.exchange_file
+            and not force
+        ):
+            raise exceptions.UserError(
+                _(
+                    "Exchange record ID=%d is not in draft state "
+                    "and has already an output value."
+                )
+                % exchange_record.id
+            )
+        if not exchange_record.direction != "outbound":
+            raise exceptions.UserError(
+                _("Exchange record ID=%d is not file is not meant to b generated")
+                % exchange_record.id
+            )
+        if exchange_record.exchange_file:
+            raise exceptions.UserError(
+                _("Exchabge record ID=%d already has a file to process!")
+                % exchange_record.id
+            )
+
+    def _generate_output(self, exchange_record, **kw):
+        """To be implemented"""
+        raise NotImplementedError()
```

Comment:
Maybe it would be good to propose something here.
I have three options on mind:

* Have the backend/type a field that marks a component of usage "generate"
* Have the backend/type a field that marks a function
* Have a selector that allows to define which options of below is used. What do you think?

Reply:
we could look up for a component for the current record model with a usage like `edi.output.{backend_type}.{exchange_type}` and fallback to `edi.output.{backend_type}` if not found.

Anyway, in the context of this PR is not required IMO.

Reply:
Leaving to round 2 then :wink: